src/dotty/tools/dotc/core/SymDenotations.scala

```diff
@@ -1060,6 +1060,12 @@ object SymDenotations {
       else defn.RootClass
     }
 
+    /** Is this denotation public ? */
+    def isPublic(implicit ctx: Context): Boolean =
+      // private-qualified denotations do not have the Private flag set,
+      // see `ParserCommon#normalize`
+      !this.is(Protected | Private) && !privateWithin.exists
```

Comment:
What you're saing that in your sbt-specific phase you have a different notion of public. 
For all other phases notion of `public` is simple: can a method\symbol\type be accessed from classes that are not subclasses of this class. Method that is `private[package]` is public from this definition. 

I'm against such change to either backend or SymDenotations.scala.